gitlabform/processors/project/files_processor.py

```diff
@@ -17,6 +17,25 @@
 
 
 class FilesProcessor(AbstractProcessor):
+    def pre_protection_check(self, project_and_group: str, configuration: dict):
+        all_branches = self.gitlab.get_branches(project_and_group)
+        sorted_branches = sorted(all_branches)
+        current_protected_state = sorted_branches["protected"]
+
+        desired_protected_state = True
+        desired_push_access_level = "no one"
+        desired_merge_access_level = "maintainers"
+
+        needs_unprotect = current_protected_state and (
```

Comment:
if this bit of logic could be extracted you might be able to add a new unit test file for files_processor and unit test it. I think this is the chunkiest bit of logic here, so would be good to add some unit tests to help document it and test logic flow makes sense